src/components/Personalization/index.js

```diff
@@ -43,41 +46,42 @@ const createPersonalization = ({ config, logger, eventManager }) => {
     storeClickMetrics
   } = createClickStorage();
   const getPageLocation = createGetPageLocation({ window });
-  const viewCache = createViewCacheManager();
-  const modules = initDomActionsModules(storeClickMetrics);
-  const executeDecisions = createExecuteDecisions({
+  const viewCache = createViewCacheManager({ createProposition });
+  const modules = initDomActionsModules();
+
+  const preprocess = createPreprocess([remapHeadOffers, remapCustomCodeOffers]);
+
+  const noOpHandler = () => undefined;
+  const domActionHandler = createDomActionHandler({
+    next: noOpHandler,
     modules,
-    logger,
-    executeActions
+    storeClickMetrics,
+    preprocess
   });
-  const handleRedirectDecisions = createRedirectHandler({
-    collect,
-    window,
-    logger,
-    showContainers
+  const measurementSchemaHandler = createMeasurementSchemaHandler({
+    next: domActionHandler
   });
-  const autoRenderingHandler = createAutorenderingHandler({
-    viewCache,
-    executeDecisions,
-    showContainers,
-    collect
+  const redirectHandler = createRedirectHandler({
+    next: measurementSchemaHandler
   });
-  const applyPropositions = createApplyPropositions({
-    executeDecisions
+  const htmlContentHandler = createHtmlContentHandler({
+    next: redirectHandler,
+    modules,
+    preprocess
   });
-  const nonRenderingHandler = createNonRenderingHandler({ viewCache });
-  const responseHandler = createOnResponseHandler({
-    autoRenderingHandler,
-    nonRenderingHandler,
-    groupDecisions,
-    handleRedirectDecisions,
-    showContainers
+
```

Comment:
The handlers above are run in a chain using a linked list. Each handler can decide whether or not to run the next one. With the final handler just being a no-op function.

Reply:
With #1036, this is now done with item processors looked up via item schema.